machine-runner/src/design/event.ts

```diff
@@ -154,28 +157,23 @@ export namespace MachineEvent {
     export type MapToActyxEvent<T extends Readonly<MachineEvent.Factory.Any[]>> =
       LooseMapToActyxEvent<T>
 
-    // =====
-    type LooseMapToMachineEvent<T, ACC extends MachineEvent.Any[] = []> = T extends Readonly<
-      [Factory<infer Key, infer Payload>, ...infer Rest]
-    >
-      ? LooseMapToMachineEvent<Rest, [...ACC, MachineEvent<Key, Payload>]>
-      : ACC
-
     /**
      * Turns a subtype of MachineEvent.Factory.Any[] into MachineEvent[].
      * @example
      * MachineEvent.Factory.MapToMachineEvent<[A,B]>
      * // where A and B are MachineEvent.Factory
      * // results in [MachineEvent.Of<A>, MachineEvent.Of<B>]
      */
-    export type MapToMachineEvent<T extends Readonly<MachineEvent.Factory.Any[]>> =
-      LooseMapToMachineEvent<T>
-
-    // =====
-    type LooseMapToPayload<T, ACC extends object[] = []> = T extends Readonly<
-      [Factory<infer Key, infer Payload>, ...infer Rest]
+    export type MapToMachineEvent<
+      T extends Readonly<MachineEvent.Factory.Any[]>,
+      ACC extends MachineEvent.Any[] = [],
+    > = T extends Readonly<
+      [
+        Factory<infer Key, infer Payload>,
+        ...infer Rest extends Readonly<[Factory<infer Key, infer Payload>, ...infer Rest]>,
```

Comment:
This is weird: are the `infer` keywords after `extends` not colliding with inferring the same names earlier? This definitely will need a comment for our future selves.

Reply:
fixed in https://github.com/Actyx/machines/pull/41/commits/a1f2e1537b83ed1ba115bb33aba3bc1136eaa849